### DISCUSSION TITLE
Best way to delete a record through a relation query?

### DISCUSSION THREAD

**Discussion by IanMitchell** [prisma/prisma#3217] [Q&A] [answered]
Hello! Struggling a little bit with how best to build a query that deletes a record through a relation - I'm using the JavaScript generator. Let's say this is my data model:

```
model Group {
  id           Int              @default(autoincrement()) @id
  linkId       String           @unique
  users        User[]

  @@index([linkId])
}

model User {
  id             Int          @default(autoincrement()) @id
  group          Group        @relation(fields: [groupId], references: [id])
  groupId        Int
  email          String

  @@unique([groupId, email])
}
```

Given a situation where I have a linkId value and an email, I want to delete the corresponding User model. I thought this would end up looking something like this:

```javascript
 prisma.group.update({
  where: {
    linkId: linkIdInput,
  },
  data: {
    users: {
      delete: [
        {
          email: emailInput
        },
      ],
    },
  },
});
```

But instead of `email: emailInput` it appears I need to pass in both the email and the groupId, which I don't have immediately available. Is there a cleaner (or more optimized) way of doing this than splitting it up into an initial `group.findOne` call followed by a `user.delete`? I had thought that connecting it this way would implicitly connect the models via `groupId` but it doesn't appear that's true.

**Comment by ryands17** [comment, marked as answer]
Hey @IanMitchell 👋 
How about this:

```ts
prisma.group.update({
    where: {
      linkId: 'something',
    },
    data: {
      users: {
        disconnect: {
          id: 1,
        },
      },
    },
  })
```

I am disconnecting the relation here (this wouldn't delete a user, just remove them from the group)
Is this what you were looking for?

**Reply by IanMitchell** [reply]
I don’t think so - since I don’t have the ID, I would have to run a second query to get it right? That puts me in the same situation as a findOne -> delete chain. Is there a reason there isn’t a more robust where chain for delete methods? It seems like the options are pretty limited

**Reply by ryands17** [reply]
If you don't have the users' ID then I think the only way to do this then would be two queries or using a raw query via `prisma.queryRaw`.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
